src/test/staking/block_reward_validator_tests.cpp

```diff
@@ -11,6 +11,33 @@
 
 namespace {
 
+class FinalizationRewardLogicMock : public proposer::FinalizationRewardLogic {
```

Comment:
Maybe this can be moved to the `unit_e_mocks.h` header file and follow the pattern outlined there?

Reply:
Moved and refactored it.